### PR TITLE
Part 3

### PR DESCRIPTION
**Pull request by laila-tk**
- Improved performance for Get Inbound Order by Warehouse Id Endpoint  by reducing the number of calls to database via revised SQL query in Stock Repo.
- Handled edge case exceptions and ensured that all tests pass for this Endpoint

### CODE REVIEW

File: ShipIt/Models/ApiModels/OrderSegment.cs [hunk 1]
```diff
@@ -15,6 +17,66 @@ public class InboundOrderResponse
...
+        public string PostalCode { get; set; }
+
+        [DatabaseColumnName("gln_addr_city")]
+        public string City { get; set; }
+
+        [DatabaseColumnName("contact_tel")]
+        public string Tel { get; set; }
+
+        [DatabaseColumnName("contact_mail")]
+        public string Mail { get; set; }
```
Comment (joeyGold1) [inline ShipIt/Models/ApiModels/OrderSegment.cs:75]: This is a lot of repeated code. Instead of having this class inherit from `DataModel`, we can use encapsulation and have the class be made from a company, stock and product model. You can use their constructors that use the `IDataReader` to construct each of them from within this class' constructor

File: ShipIt/Repositories/StockRepository.cs [hunk 2]
```diff
@@ -59,6 +60,16 @@ public Dictionary<int, StockDataModel> GetStockByWarehouseAndProductIds(int ware
...
         }
+
+
+        public Dictionary<int, StockProduct> GetStockByWarehouseIdWithProducts(int warehouseId)
+        {            
+            string sql = string.Format("SELECT stock.w_id, gtin.p_id, gtin.gtin_cd,gtin.gtin_nm, stock.hld, gtin.l_th, gtin.min_qt , gcp.* FROM stock join gtin on stock.p_id = gtin.p_id join gcp on gcp.gcp_cd = gtin.gcp_cd where stock.w_id = @w_id and stock.hld < gtin.l_th and gtin.ds = 0");
+            var parameter = new NpgsqlParameter("@w_id", warehouseId);
+            string noProductWithIdErrorMessage = string.Format("No stock found with w_id: @w_id");
+            var stock = base.RunGetQuery(sql, reader => new StockProduct(reader), noProductWithIdErrorMessage, parameter);
+            return stock.ToDictionary(s => s.ProductId, s => s);
```
Comment (joeyGold1) [inline ShipIt/Repositories/StockRepository.cs:71]: Nice query! This is complicated stuff!

**Review by joeyGold1 — COMMENTED** [review]
Nice work! Do all the tests still pass? How much faster is the request now than it was before?
